wiki/plugins/macros/mdx/macro.py

```diff
@@ -1,106 +1,255 @@
-                                value = True
+
+                    args = m.group('args')
+
+                    if args:
+
+                        args_list = []
+
+                        for arg in ARG_RE.finditer(args):
+
+                            value = arg.group('value')
+
+                            #if value is None:
+
+                                #value = True
+
                             if isinstance(value, string_types):
+
                                 # If value is enclosed with ': Remove and
+
                                 # remove escape sequences
+
                                 if value.startswith("'") and len(value) > 2:
+
                                     value = value[1:-1]
+
                                     value = value.replace("\\\\", "¤KEEPME¤")
+
                                     value = value.replace("\\", "")
+
                                     value = value.replace("¤KEEPME¤", "\\")
-                            kwargs_dict[str(arg)] = value
-                        line = getattr(self, macro)(**kwargs_dict)
+
+                            if value is not None:
+
+                                args_list.append(str(len(args_list)))
+
+                                args_list.append(value)
+
+                        line = getattr(self, macro)(*args_list)
+
                     else:
+
                         line = getattr(self, macro)()
+
             if line is not None:
+
                 new_text.append(line)
+
         return new_text
 
+
+
     def article_list(self, depth="2"):
+
         html = render_to_string(
+
             "wiki/plugins/macros/article_list.html",
+
             context={
+
                 'article_children': self.markdown.article.get_children(
+
                     article__current_revision__deleted=False),
+
                 'depth': int(depth) + 1,
+
             })
+
         return self.markdown.htmlStash.store(html, safe=True)
+
     article_list.meta = dict(
+
         short_description=_('Article list'),
+
         help_text=_('Insert a list of articles in this level.'),
+
         example_code='[article_list depth:2]',
+
         args={'depth': _('Maximum depth to show levels for.')}
+
     )
 
+
+
     def toc(self):
+
         return "[TOC]"
+
     toc.meta = dict(
+
         short_description=_('Table of contents'),
+
         help_text=_('Insert a table of contents matching the headings.'),
+
         example_code='[TOC]',
+
         args={}
+
     )
 
+
+
     def wikilink(self):
+
         return ""
+
     wikilink.meta = dict(
+
         short_description=_('WikiLinks'),
+
         help_text=_(
+
             'Insert a link to another wiki page with a short notation.'),
+
         example_code='[[WikiLink]]',
+
         args={})
+
+    
+
+    def p(self, *args):
+
+        if len(args)>1:
+
+            prep, construal = args[0], args[1]
+
+            return '['+prep+'](/'+prep+'/'+construal+')'
```

Comment:
This doesn't actually check that the 2nd arg is a construal: it could be a supersense (or name of any top-level article). Should `[p from Source]` be shorthand for `[p from Source--Source]`? If so, check if `args[1]` contains `--`, and if not, set `construal = args[1] + '--' + args[1]`? Unless `args[1]` starts with a backtick, I guess, to support `$ etc.

Reply:
I can add this in.